palette/src/lib.rs

```diff
@@ -42,53 +42,49 @@
-//! 2. Its **reference white** or **white point**
-//!
-//! The **primaries** together represent the total *gamut* (i.e. displayable
-//! range of colors) of that color space, while the **white point** defines
-//! which concrete tristimulus value corresponds to a real, physical white
-//! reflecting object being lit by a known light source and observed by the
-//! 'standard observer' (i.e. a standardized model of human color perception).
-//!
-//! The informal "RGB" color space is such a tristimulus color space, since
-//! it is defined by three values, but it is underspecified since we don't
-//! know which primaries are being used (i.e. how exactly are the canonical
-//! "red", "green", and "blue" defined?), nor its white point. In most cases,
-//! when people talk about "RGB" or "Linear RGB" colors, what they are
-//! *actually* talking about is the "Linear sRGB" color space, which uses the
-//! primaries and white point defined in the sRGB standard, but which *does
-//! not* have the (non-linear) sRGB *transfer function* applied.
-//!
-//! This library takes these things into account, and attempts to provide an
-//! interface which will let those who don't care so much about the intricacies
-//! of color still use colors correctly, while also allowing the advanced user
-//! a high degree of flexibility in how they use it.
+//! See the [rgb] module for a deeper dive into RGB and (non-)linearity.
+//!
+//! # Color Spaces
+//!
+//! "RGB" and other tristimulus based spaces like CIE Xyz are probably the most
+//! widely known color spaces. These spaces are great when you want to perform
+//! physically correct math on color (like in a 2d or 3d rendering program) but
+//! there are also color spaces that are not defined in terms of tristimulus
+//! values.
+//!
+//! You have probably used a color picker with a rainbow wheel and a brightness
+//! slider. That may have been an HSV or an HSL color picker, where the color is
+//! encoded as hue, saturation and brightness/lightness. Even though these
+//! spaces are defined using 3 values, they *aren't* based on tristimulus
+//! values, since those three values don't have a direct relation to human
+//! vision (i.e. our S, M, and L cones, as discussed in the previous section).
+//! Such color spaces are excellent when it comes to humans intuitively
+//! selecting color values, though, and as such are the go-to choice when this
+//! interaction is needed. They can then be converted into other color spaces in
+//! order to actually perform modifications to them
+//!
+//! There's also a group of color spaces that are designed to be perceptually
+//! uniform, meaning that the perceptual change is equal to the numerical
+//! change. An example of this is the CIE L\*a\*b\* color space. These color
+//! spaces are excellent when you want to "blend" between colors in a
+//! *perceptually pleasing* manner (for example, in a data visualization) rather
+//! than a *physically correct* one.
```

Comment:
I think the idea is that it doesn't linearly match actual light intensity, but rather how we perceive the color (like green looking more intense than blue), but it can surely be clarified. And I interpret the "data visualization" part as meaning that it's suitable for generating colors for graphs and such, since the generated colors are easier to make uniform. It may be less confusing to just remove that example.

Reply:
I replaced the last sentence with "These color spaces are excellent when you want to "blend" between colors in a *perceptually pleasing* manner rather than based on *physical light intensity*.", hoping it gets the point across. This whole section need to be rewritten later. Maybe have a section for each color space group and summarize their shared properties.